lib/jsonapi-serializers/serializer.rb

```diff
@@ -260,12 +268,18 @@ def self.serialize(objects, options = {})
       includes = options[:include]
       includes = (includes.is_a?(String) ? includes.split(',') : includes).uniq if includes
 
+      fields = options[:fields] || {}
+      fields = Hash[fields.map do |type, whitelisted_fields|
```

Comment:
This could be cleaned up and commented a bit, it's very non-obvious what it's doing by looking at it.